### PR TITLE
Hook-up student orgs list and detail views

### PR DESCRIPTION
**Pull request by drewvolz**
* Convert to functional components
* Use hooks for navigation

Recommended to [view without whitespace ](https://github.com/StoDevX/AAO-React-Native/pull/5976/files?w=1)

---

~|~
--|--
ESLint | - 344 problems (0 errors, 344 warnings)<br>+ 338 problems (0 errors, 338 warnings)
TSC | - 303 lines <br> + 295 lines
TSC +- | -10 TS7006<br>+2 TS7006

### CODE REVIEW

**Comment by hawkrives** [comment]
Don't forget a "make files" in the root of the repo!

**Comment by drewvolz** [comment]
Ah, thanks. Maybe I should add that as a hook locally.

**Comment by hawkrives** [comment]
@drewvolz I pushed the regenerated error files

File: source/views/student-orgs/detail.ts [hunk 1]
```diff
@@ -1,7 +1,17 @@
 import {Platform} from 'react-native'
 
-import {StudentOrgsDetailView as IosDetail} from './detail-ios'
-import {StudentOrgsDetailView as AndroidDetail} from './detail-android'
+import {
+	View as IosDetail,
+	NavigationOptions as ListNavigationOptions,
```
Comment (hawkrives) [inline source/views/student-orgs/detail.ts]:
```suggestion
	NavigationOptions as IosDetailNavigationOptions,
```

maybe?

File: source/views/student-orgs/detail.ts [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
+} from './detail-ios'
+import {
+	View as AndroidDetail,
+	NavigationOptions as DetailNavigationOptions,
```
Comment (hawkrives) [inline source/views/student-orgs/detail.ts]:
```suggestion
	NavigationOptions as AndroidDetailNavigationOptions,
```

File: source/views/student-orgs/detail.ts [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
+} from './detail-android'
 
-export const StudentOrgsDetailView =
-	Platform.OS === 'ios' ? IosDetail : AndroidDetail
+const StudentOrgsDetailView = Platform.OS === 'ios' ? IosDetail : AndroidDetail
+
+export {StudentOrgsDetailView as View}
+
+export const NavigationOptions =
+	Platform.OS === 'ios' ? ListNavigationOptions : DetailNavigationOptions
```
Comment (hawkrives) [inline source/views/student-orgs/detail.ts]:
```suggestion
	Platform.OS === 'ios' ? IosDetailNavigationOptions : AndroidDetailNavigationOptions
```

**Review by hawkrives — CHANGES_REQUESTED** [review]
maybe tweak these names to make it clear they're platform-dependent?

**Comment by hawkrives** [comment]
LGTM! What do you think about tweaking the names in that one file?

**Comment by drewvolz** [comment]
Oh no. I made that change but it's not here. Where did that commit go...

**Review by hawkrives — APPROVED** [review]

**Review by rye — APPROVED** [review]

**Pull request merged**